Challenge 1 - Update StateOfTexas.mjs

PriDhingra opened the pull request.

PriDhingra committed:
Challenge 1 - Update StateOfTexas.mjs

PriDhingra committed:
Challenge 1 - Update StateOfTexas.mjs

PriDhingra committed:
Update StateOfTexas.js

1. Handling errors using try-catch.
2. More readable code.
3. Assigning the task of manipulating data to a different function
4. Handling multiple fetch requests in parallel to reduce the run time.

Nefsen402 commented on scrapers/StateOfTexas.mjs:
```diff
@@ -1,32 +1,76 @@
 export default ({ database, DataScraper }) => {
 	return DataScraper(database, 'StateOfTexas', async () => {
+        //Initialize an empty array to store result
 		const data = [];
```
Please be mindful about the code indentation here.

Nefsen402 reviewed (changes requested):
Great work! Nice idea on splitting some of the parsing into a separate function! Better yet, some of the fetches could go in there as well.

PriDhingra committed:
Update StateOfTexas.mjs

Some tweaks and changes to the code - 
1. Proper Indentations and remove white spaces.
2. Reuse baseUrl.
3. fetch/JSON parsing happens in one step in series instead of in parallel.
